pkg/controller/plan/adapter/ovirt/client.go

```diff
@@ -76,23 +92,28 @@ func (r *Client) CheckSnapshotReady(vmRef ref.Ref, snapshot string) (ready bool,
 		err = liberr.Wrap(err)
 		return
 	}
-	jobs, err := r.getJobs(correlationID)
+	events, err := r.getEvents(correlationID)
 	if err != nil {
 		err = liberr.Wrap(err)
 		return
 	}
-	if len(jobs) < 1 {
-		err = liberr.New("No jobs found for correlation ID", "correlationID", correlationID)
+	if len(events) < 1 {
+		err = liberr.New("No event found for correlation ID", "correlationID", correlationID)
 		return
 	}
-	ready = true
-	for _, job := range jobs {
-		switch job.MustStatus() {
-		case ovirtsdk.JOBSTATUS_FAILED, ovirtsdk.JOBSTATUS_ABORTED:
+	for _, event := range events {
+		code, exists := event.Code()
+		if !exists {
+			err = liberr.New("The event does not have a code.", "event", event)
+			return
```

Comment:
it shouldn't really happen that we get an event without a code from ovirt but if we have this check, and we ever get here then we better continue in this case